pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -111,37 +140,42 @@ func (p *Propagator) addDependentToOwners(n *node, owners []metatypes.OwnerRefer
 					OwnerReference: owner,
 					Namespace:      n.identity.Namespace,
 				},
-				dependents: make(map[*node]struct{}),
+				dependentsLock: &sync.RWMutex{},
+				dependents:     make(map[*node]struct{}),
 			}
-			p.uidToNode[ownerNode.identity.UID] = ownerNode
+			p.uidToNode.Write(ownerNode.identity.UID, ownerNode)
 			p.gc.dirtyQueue.Add(ownerNode)
 		}
+		ownerNode.dependentsLock.Lock()
+		defer ownerNode.dependentsLock.Unlock()
```

Comment:
Nit: this keeps the node locked longer than necessary. not a big deal. But I'd still make these three lines a method of ownerNode.